### PR TITLE
chore: switch recording filtering to be a query

### PR DESCRIPTION
**Pull request by daibhin**
## Problem

Recordings are fetched using a custom lookup endpoint right now. We want to move to to a typed lookup `Query` so that:
1. We get validation between the frontend & backend
2. (In future) we can cache query results to load replays faster

## Changes

1. Split log entries into a separate property type
    - Allows us to do really advanced message queries e.g. regex matches
    - AND / OR of log types and messages
2. Rename the previous `RecordingFilters` to `LegacyRecordingFilters`
  - They can largely be ignored now but exist as typed references for converting URLs containing old filters to the new schema
3. Move the current filters to a new `RecordingsQuery` that is defined in the schema
4. Abstract some shared methods from HogQL code

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

There should be no functional changes e.g. existing tests continue to pass (with changes to match the new schema)

### CODE REVIEW

File: posthog/hogql/property.py [hunk 1]
```diff
@@ -156,54 +160,57 @@ def _handle_bool_values(value: ValueT, field: ast.Field, property: Property, tea
     return value
 
 
-def _field_to_compare_op(
```
Comment (daibhin) [inline posthog/hogql/property.py:159]: I might pull these changes into a separate PR and get approval from Product Analytics

File: posthog/session_recordings/queries/session_recording_list_from_filters.py [hunk 2]
```diff
@@ -258,39 +235,8 @@ def _where_predicates(self) -> Union[ast.And, ast.Or]:
...
-            )
-            exprs.append(
-                ast.CompareOperation(
-                    op=op,
-                    left=ast.Call(name="argMinMerge", args=[ast.Field(chain=["s", "snapshot_source"])]),
-                    right=ast.Constant(value=self._filter.snapshot_source_filter.value),
-                ),
-            )
-
-        return ast.And(exprs=exprs) if exprs else ast.Constant(value=True)
```
Comment (daibhin) [inline posthog/session_recordings/queries/session_recording_list_from_filters.py:293]: Being able to get rid of all this kind of custom filtering code is the main intention behind this PR. Everything is now handled inside the single `property_to_expr` method

File: posthog/hogql/property.py [hunk 3]
```diff
@@ -275,7 +275,7 @@ def property_to_expr(
         | DataWarehousePersonPropertyFilter
     ),
     team: Team,
-    scope: Literal["event", "person", "session", "replay", "replay_entity", "replay_pdi"] = "event",
```
Comment (daibhin) [inline posthog/hogql/property.py:278]: Was never used anywhere

**Review by marandaneto — APPROVED** [review]
trusting the tests and that `property_to_expr` does the job 😀

**Comment by daibhin** [comment]
> trusting the tests and that property_to_expr does the job 😀

@marandaneto so am I tbh 😂 I joke, there's a decent amount of tests covering this file that I'm confident it won't break the primary flows at the very least. And we know there's some bugs in more niche cases (2 event with property filters)

**Pull request merged**